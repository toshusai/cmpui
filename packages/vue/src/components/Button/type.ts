export const ButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type ButtonVariant = (typeof ButtonVariant)[keyof typeof ButtonVariant];

export const ButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type ButtonSize = (typeof ButtonSize)[keyof typeof ButtonSize];

export type ButtonProps = {
  variant: ButtonVariant;
  size: ButtonSize;
  disabled: boolean;
};
