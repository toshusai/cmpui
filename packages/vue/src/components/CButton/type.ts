export const CButtonVariant = {
  primary: "primary",
  secondary: "secondary",
} as const;

export type CButtonVariant =
  (typeof CButtonVariant)[keyof typeof CButtonVariant];

export const CButtonSize = {
  S: "S",
  M: "M",
  L: "L",
} as const;

export type CButtonSize = (typeof CButtonSize)[keyof typeof CButtonSize];

export type CButtonProps = {
  variant?: CButtonVariant;
  size?: CButtonSize;
  disabled?: boolean;
};
