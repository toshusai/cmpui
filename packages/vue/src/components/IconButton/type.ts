export const IconButtonSize = {
  S: "S",
  M: "M",
} as const;

export type IconButtonSize =
  (typeof IconButtonSize)[keyof typeof IconButtonSize];

export type IconButtonProps = {
  bordered?: boolean;
  disabled?: boolean;
  rounded?: boolean;
  selected?: boolean;
  size?: IconButtonSize;
};
