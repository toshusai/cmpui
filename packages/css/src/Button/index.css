.cmpui_button {
  background-color: var(--cmpui-primary-color);

  align-items: center;
  border-radius: 9999px;
  border: none;
  box-sizing: border-box;
  color: white;
  cursor: pointer;
  display: flex;
  font-family: unset;
  font-size: 16px;
  height: 32px;
  justify-content: center;
  line-height: 16px;
  outline: none;
  padding: 0;
  padding: 8px 16px;
  text-align: center;
  text-decoration: none;
}

.cmpui_button:hover {
  background-color: var(--cmpui-primary-hover-color);
}

.cmpui_button:active {
  background-color: var(--cmpui-primary-active-color);
}

.cmpui_button:focus-visible {
  outline: var(--cmpui-primary-color) solid 1px;
  outline-offset: 1px;
}

/* disabled */

.cmpui_button:disabled {
  cursor: default;
  opacity: 0.5;
}

.cmpui_button:disabled:hover {
  background-color: var(--cmpui-primary-color);
}

.cmpui_button:disabled:active {
  background-color: var(--cmpui-primary-color);
}

/* Variant: secondary */

.cmpui_button[data-variant="secondary"] {
  background-color: var(--cmpui-secondary-color);
}

.cmpui_button[data-variant="secondary"]:hover {
  background-color: var(--cmpui-secondary-hover-color);
}

.cmpui_button[data-variant="secondary"]:active {
  background-color: var(--cmpui-secondary-active-color);
}

.cmpui_button[data-variant="secondary"]:disabled:hover {
  background-color: var(--cmpui-secondary-color);
}

.cmpui_button[data-variant="secondary"]:disabled:active {
  background-color: var(--cmpui-secondary-color);
}

/* Size: S */

.cmpui_button[data-size="S"] {
  font-size: 12px;
  height: 24px;
  line-height: 12px;
  padding: 4px 12px;
}
