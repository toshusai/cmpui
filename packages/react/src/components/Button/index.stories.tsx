import type { Meta, StoryObj } from "@storybook/react";

import { Button } from ".";

const meta: Meta<typeof Button> = {
  title: "Actions/Button",
  component: Button,
};

export default meta;
type Story = StoryObj<typeof Button>;

export const Basic: Story = {
  render: () => {
    return <Button>Button</Button>;
  },
};

export const Secondary: Story = {
  render: () => {
    return <Button variant="secondary">Button</Button>;
  },
};

export const Small: Story = {
  render: () => {
    return <Button size="S">Button</Button>;
  },
};

export const Disabled: Story = {
  render: () => {
    return <Button disabled>Button</Button>;
  },
};
