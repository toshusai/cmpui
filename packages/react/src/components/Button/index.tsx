import { forwardRef } from "react";
import { CustomJSXElement } from "../../types/CustomJSXElement";
import { classNames } from "../../utils/classNames";

export type ButtonProps<T extends CustomJSXElement = "button"> = {
  as?: T;
  size?: "S" | "M";
  variant?: "primary" | "secondary";
} & Omit<React.ComponentProps<T>, "as">;

export const Button = forwardRef(function Button<T extends CustomJSXElement>(
  {
    variant = "primary",
    size = "M",
    children,
    className,
    as,
    ...props
  }: ButtonProps<T>,
  ref: React.ForwardedRef<HTMLButtonElement>,
) {
  const Component = as || "button";
  return (
    <Component
      ref={ref}
      type="button"
      className={classNames("cmpui_button", className)}
      {...props}
      data-variant={variant}
      data-size={size}
    >
      {children}
    </Component>
  );
}) as <T extends CustomJSXElement = "button">(p: ButtonProps<T>) => JSX.Element;
