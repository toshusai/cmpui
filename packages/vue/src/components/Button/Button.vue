<script setup lang="ts">
import { ButtonProps } from "./type";

withDefaults(defineProps<ButtonProps>(), {
  variant: "primary",
  size: "M",
  disabled: false,
});
</script>

<template>
  <button
    class="cmpui_button"
    :data-variant="variant"
    :data-size="size"
    :disabled="disabled"
  >
    <slot></slot>
  </button>
</template>
