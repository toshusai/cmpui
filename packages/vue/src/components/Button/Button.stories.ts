import type { Meta, StoryObj } from "@storybook/vue3";
import Button from "./Button.vue";

const meta = {
  component: Button,
  tags: ["autodocs"],
} satisfies Meta<typeof Button>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    default: "Button",
    variant: "primary",
  },
};

export const Secondary: Story = {
  args: {
    default: "Button",
    variant: "secondary",
  },
};

export const Small: Story = {
  args: {
    default: "Button",
    size: "S",
  },
};

export const Disabled: Story = {
  args: {
    default: "Button",
    disabled: true,
  },
};
