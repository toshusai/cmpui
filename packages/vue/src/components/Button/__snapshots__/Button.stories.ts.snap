// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button Default smoke-test 1`] = `
<button class="cmpui_button"
        data-variant="primary"
        data-size="M"
        default="Button"
>
  Button
</button>
`;

exports[`components/Button Disabled smoke-test 1`] = `
<button class="cmpui_button"
        data-variant="primary"
        data-size="M"
        disabled
        default="Button"
>
  Button
</button>
`;

exports[`components/Button Secondary smoke-test 1`] = `
<button class="cmpui_button"
        data-variant="secondary"
        data-size="M"
        default="Button"
>
  Button
</button>
`;

exports[`components/Button Small smoke-test 1`] = `
<button class="cmpui_button"
        data-variant="primary"
        data-size="S"
        default="Button"
>
  Button
</button>
`;
